x-pack/plugins/cases/public/components/custom_fields/custom_fields_list/index.test.tsx

```diff
@@ -54,15 +42,79 @@ describe('CustomFieldsList', () => {
   });
 
   it('shows single CustomFieldsList correctly', async () => {
-    appMockRender.render(<CustomFieldsList customFields={[customFieldsMock[0]]} />);
+    appMockRender.render(
+      <CustomFieldsList
+        customFields={[customFieldsConfigurationMock[0]]}
+        onDeleteCustomField={onDeleteCustomField}
+      />
+    );
 
-    expect(screen.getByTestId('droppable')).toBeInTheDocument();
+    const droppable = screen.getByTestId('droppable');
+
+    expect(droppable).toBeInTheDocument();
     expect(screen.getAllByTestId('draggable').length).toEqual(1);
+    expect(
+      within(droppable).getByTestId(`${customFieldsConfigurationMock[0].key}-custom-field-delete`)
+    ).toBeInTheDocument();
   });
 
   it('does not show droppable field when no custom fields', () => {
-    appMockRender.render(<CustomFieldsList customFields={[]} />);
+    appMockRender.render(
+      <CustomFieldsList customFields={[]} onDeleteCustomField={onDeleteCustomField} />
+    );
 
     expect(screen.queryByTestId('droppable')).not.toBeInTheDocument();
   });
+
+  it('shows confirmation modal when deleting a field ', async () => {
+    appMockRender.render(<CustomFieldsList {...props} />);
+
+    const droppable = screen.getByTestId('droppable');
+
+    userEvent.click(
+      within(droppable).getByTestId(`${customFieldsConfigurationMock[0].key}-custom-field-delete`)
+    );
+
+    await waitFor(() => {
+      expect(screen.getByTestId('confirm-delete-custom-field-modal')).toBeInTheDocument();
+    });
+  });
+
+  it('calls onDeleteCustomField when confirm', async () => {
+    appMockRender.render(<CustomFieldsList {...props} />);
+
+    const droppable = screen.getByTestId('droppable');
+
+    userEvent.click(
+      within(droppable).getByTestId(`${customFieldsConfigurationMock[0].key}-custom-field-delete`)
+    );
+
+    await waitFor(() => {
+      expect(screen.getByTestId('confirm-delete-custom-field-modal')).toBeInTheDocument();
+    });
+
+    userEvent.click(screen.getByText('Delete'));
+    await waitFor(() => {
+      expect(onDeleteCustomField).toHaveBeenCalledWith(customFieldsConfigurationMock[0].key);
+    });
+  });
+
+  it('does not call onDeleteCustomField when cancel', async () => {
+    appMockRender.render(<CustomFieldsList {...props} />);
+
+    const droppable = screen.getByTestId('droppable');
+
+    userEvent.click(
+      within(droppable).getByTestId(`${customFieldsConfigurationMock[0].key}-custom-field-delete`)
+    );
+
+    await waitFor(() => {
+      expect(screen.getByTestId('confirm-delete-custom-field-modal')).toBeInTheDocument();
+    });
+
+    userEvent.click(screen.getByText('Cancel'));
+    await waitFor(() => {
+      expect(onDeleteCustomField).not.toHaveBeenCalledWith();
```

Comment:
I think is better to wait here so that the `confirm-delete-custom-field-modal` is not shown anymore. This way we can be sure that `expect(onDeleteCustomField).not.toHaveBeenCalledWith();` is not true because it has not been called yet.